### PR TITLE
Convert all tests to python

### LINKED ISSUE #189: Convert unit tests to python

### PR DESCRIPTION
**Pull request by philwinder**
Fixes #189 #190 #191

### CODE REVIEW

**Comment by philwinder** [comment]
Fixes #189 #190 #191

File: testing/util/Golang.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    def test(self):
+        script_dir = os.path.dirname(os.path.realpath(__file__))
+        code_dir = script_dir + "/../../sockshop/" + self.container_dir
+
+        test_container_name = self.container_dir + '-test'
+        docker = Docker()
+        docker.execute(['docker', 'build', '-t', test_container_name, code_dir])
+        docker.execute(['docker', 'run', '--rm', test_container_name, 'go', 'test',
+                        'github.com/weaveworks/weaveDemo/' + self.container_dir])
```
Comment (idcrosby) [inline testing/util/Golang.py:18]: out of scope for this PR, but we should change all references of weaveDemo to microservices-demo

**Comment by idcrosby** [comment]
Why do we run the unit tests in containers? It feels like unnecessary work and potential for issues (hard coded paths, etc).

**Comment by idcrosby** [comment]
The tests run fine on my machine (using docker for mac). I like the setup.

**Comment by idcrosby** [comment]
LGTM

**Pull request merged**